src/Avalonia.Controls/Platform/ISandboxBookmark.cs

```diff
@@ -0,0 +1,56 @@
+using System;
+
+namespace Avalonia.Platform
+{
+    /// <summary>
+    /// Factory to create Mac App Store Sandbox Bookmark from previously saved bookmark data.
+    /// Use only in macOS
+    /// </summary>
+    public interface ISandboxBookmarkFactory
+    {
+        /// <summary>
+        /// Create Mac App Store Sandbox Bookmark from previously saved <see cref="ISandboxBookmark.BookmarkData"/>
+        /// </summary>
+        ISandboxBookmark Create(byte[] bookmarkData);
+    }
+
+    /// <summary>
+    /// Mac App Store Sandbox Bookmark - an object that allows to access files outside of sandbox
+    /// </summary>
+    public interface ISandboxBookmark
+    {
+        /// <summary>
+        /// File Path of the Bookmark
+        /// </summary>
+        string Url { get; }
```

Comment:
Type "string" but name is "Url". Should it be "Uri" type with "file://" protocol?

Reply:
Though, nevermind, we already have the same in https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Controls/UrlOpenedEventArgs.cs